pkg/cluster/cluster.go

```diff
@@ -332,21 +331,65 @@ func (c *Cluster) Create() (err error) {
 			err = fmt.Errorf("could not sync pgbackrest config: %v", err)
 			return err
 		}
-		c.logger.Info("a pgbackrest config has been successfully synced")
 		if c.Postgresql.Spec.Backup.Pgbackrest.Restore.ID != c.Status.PgbackrestRestoreID {
 			if err = c.syncPgbackrestRestoreConfig(); err != nil {
 				err = fmt.Errorf("could not sync pgbackrest restore config: %v", err)
 				return err
 			}
 			c.KubeClient.SetPgbackrestRestoreCRDStatus(c.clusterName(), c.Postgresql.Spec.Backup.Pgbackrest.Restore.ID)
 			c.Status.PgbackrestRestoreID = c.Postgresql.Spec.Backup.Pgbackrest.Restore.ID
-			c.logger.Info("a pgbackrest restore config has been successfully synced")
 		} else {
 			if err = c.createPgbackrestRestoreConfig(); err != nil {
 				err = fmt.Errorf("could not create a pgbackrest restore config: %v", err)
 				return err
 			}
-			c.logger.Info("a pgbackrest restore config has been successfully created")
+		}
+		// if found then create a sts for pgbackrest repo
+		// add the config info to this repo
+		// add the config info to the pg pods about this repo
+		for _, repo := range c.Postgresql.Spec.Backup.Pgbackrest.Repos {
+			if repo.Storage == "pvc" {
+				if len(repo.Volume.Size) == 0 {
+					err = fmt.Errorf("could not create pgbackrest statefulset for pvc without size: %v", err)
+					return err
+				}
+				pvcName := c.getPgbackrestRepoHostName()
+				backupSecretName := c.getPgbackrestSecretName()
+				if err = c.createPgbackrestRepoHostConfig(); err != nil {
+					err = fmt.Errorf("could not create a pgbackrest repo-host config: %v", err)
+					return err
+				}
+				// if statefulset for pvc is already created then only mount an additional volume in it for this repo
+				err := c.createPVCSecret(backupSecretName)
```

Comment:
I found the naming confusing at first. Maybe lets rename getPgbackrestSecretName to getPgbackrestCertSecretName and createPVCSecret to createPgbackrestCertSecret